src/include/btree.i

```diff
@@ -530,6 +530,9 @@ __wt_page_modify_set(WT_SESSION_IMPL *session, WT_PAGE *page)
 		WT_FULL_BARRIER();
 	}
 
+	if (S2C(session)->modified == 0)
+		S2C(session)->modified = 1;
+
```

Comment:
Consider moving this up inside the `if` before setting `S2BT(session)->modified = 1`.

Or is there some case where the btree is dirty on entry but the connection could be clean?